src/RecoveryServices/RecoveryServices.Backup.Autorest/custom/Cmdlets/Policy/Edit-AzrecoveryServicesBackupRetentionPolicyClientObject.ps1

```diff
@@ -0,0 +1,945 @@
+                      $errormsg="Daily retention can't be enabled with weekly backup schedule'."
+                      throw $errormsg
+                  }
+                  if ($policyObject.RetentionPolicy.DailySchedule.RetentionDuration.Count -ne $null) 
+                  {
+                      # DailySchedule exists, check retentionDuration
+                      if ( $DailyRetentionDurationInDays -ne $null ) 
+                      {
+                         $policyObject.RetentionPolicy.DailySchedule.RetentionDuration.Count = $DailyRetentionDurationInDays 
+                      }
+                  }
+                  elseif( $policyObject.RetentionPolicy.DailySchedule.RetentionDuration.Count -eq $null )
+                  {
+                     # DailySchedule doesn't exist, create a new one
+                     $policyObject.RetentionPolicy.DailySchedule = [Microsoft.Azure.PowerShell.Cmdlets.RecoveryServices.Models.Api20230201.DailyRetentionSchedule]::new()
+                     $policyObject.RetentionPolicy.DailySchedule.retentionDuration = [Microsoft.Azure.PowerShell.Cmdlets.RecoveryServices.Models.Api20230201.RetentionDuration]::new()
+                     $policyObject.RetentionPolicy.DailySchedule.RetentionDuration.Count = [System.Nullable[int]]::new($DailyRetentionDurationInDays)
+                     $policyObject.RetentionPolicy.dailySchedule.retentionDuration.durationType = "Days"
+                     $policyObject.RetentionPolicy.DailySchedule.RetentionTime=$policyObject.SchedulePolicy.ScheduleRunTime    
+                  }
+              }    
+
+              if($EnableWeeklyRetention -eq $false)
+              {
+                  if($policyObject.SchedulePolicy.ScheduleRunFrequency -eq "Weekly")
+                  {
+                      $errormsg = "Weekly retention can't be disabled'."
+                      throw $errormsg
+                  }
+                  else
+                  {
+                      $policyObject.RetentionPolicy.WeeklySchedule=$null
+                  }
+              }
+              elseif ($EnableWeeklyRetention -eq $true ) 
+              { 
+                  # Set weekly retention properties
+                  if ($policyObject.RetentionPolicy.WeeklySchedule.RetentionDuration.Count -ne $null) 
+                  {
+                      if ($WeeklyRetentionDaysOfTheWeek -ne $null)
+                      {
+                          if($policyObject.SchedulePolicy.ScheduleRunFrequency -eq "Daily")
+                          {
+                              $policyObject.RetentionPolicy.WeeklySchedule.DaysOfTheWeek=$WeeklyRetentionDaysOfTheWeek 
+                          }
+                          elseif($policyObject.SchedulePolicy.ScheduleRunFrequency -eq "Weekly")
+                          {
+                              $policyObject.RetentionPolicy.WeeklySchedule.DaysOfTheWeek=$policyObject.SchedulePolicy.ScheduleRunDay
```

Comment:
that is in case of monthly or yearly week based retention.
For weekly retention WeeklySchedule.DaysOfTheWeek can't be modified. Its value is exactly same as SchedulePolicy.ScheduleRunDay